.github/workflows/node-ci-main.yml

```diff
@@ -50,7 +50,14 @@ jobs:
     - name: Jest with coverage
       run: yarn coverage
 
-    - name: Upload Coverage
+    - name: Upload Coverage (Codecov)
+      uses: codecov/codecov-action@v3
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        files: ./coverage/coverage-final.json
+        fail_ci_if_error: true
```

Comment:
Are we sure we need to fail CI if we're not able to upload code coverage? Is there like a warn option or anything?

Reply:
I believe it used to be false... I found that because it was flaky I didn't notice that it was failing for a while.  My preference is to put it in our face and if it becomes problematic we fix it vs. limping along. :)

Reply:
We have Coveralls to fall back to if Codecov continues to be flaky.  I think one possible reason it was flaky is that we were only upload `jest` results when running on `main`.  That being said, Coveralls approach of having a separate signal to indicate when all reports have been uploaded is a better approach to our current Codecov setup.  We could make Codecov better by merging the reports ourselves before uploading them, but that requires leveraging artifacts which will end up making our workflow files even more complex.